rasa/shared/core/events.py

```diff
@@ -377,27 +409,31 @@ def _from_parse_data(
         )
 
     def __hash__(self) -> int:
+        """Returns unique hash of object."""
         return hash((self.text, self.intent_name, jsonpickle.encode(self.entities)))
 
     @property
     def intent_name(self) -> Optional[Text]:
+        """Returns intent name or `None` if no intent."""
         return self.intent.get(INTENT_NAME_KEY)
 
     def __eq__(self, other: Any) -> bool:
+        """Compares object with other object."""
         if not isinstance(other, UserUttered):
-            return False
-        else:
-            return (
-                self.text,
-                self.intent_name,
-                [jsonpickle.encode(ent) for ent in self.entities],
-            ) == (
-                other.text,
-                other.intent_name,
-                [jsonpickle.encode(ent) for ent in other.entities],
-            )
+            return NotImplemented
+
+        return (
+            self.text,
+            self.intent_name,
+            [jsonpickle.encode(ent) for ent in self.entities],
+        ) == (
+            other.text,
+            other.intent_name,
+            [jsonpickle.encode(ent) for ent in other.entities],
+        )
 
     def __str__(self) -> Text:
+        """Returns text representation of event."""
         return (
             f"UserUttered(text: {self.text}, intent: {self.intent}, "
             f"entities: {self.entities})"
```

Comment:
that will contain a dict with all the confidences, extractors, etc, should we filter only entity names here?

Reply:
see my answer to https://github.com/RasaHQ/rasa/pull/7388#discussion_r537461938